src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageServiceFactory.cs

```diff
@@ -101,14 +101,22 @@ public ITemporaryTextStorageInternal CreateTemporaryTextStorage()
         string storageName, long offset, long size, SourceHashAlgorithm checksumAlgorithm, Encoding? encoding, ImmutableArray<byte> contentHash)
         => new(this, storageName, offset, size, checksumAlgorithm, encoding, contentHash);
 
-    ITemporaryStreamStorageInternal ITemporaryStorageServiceInternal.CreateTemporaryStreamStorage()
-        => CreateTemporaryStreamStorage();
+    public TemporaryStorageHandle WriteToTemporaryStorage(Stream stream, CancellationToken cancellationToken)
+    {
+        var storage = new TemporaryStreamStorage(this);
+        storage.WriteStream(stream, cancellationToken);
+        var identifier = new TemporaryStorageIdentifier(storage.Name, storage.Offset, storage.Size);
+        return new(storage, identifier);
+    }
```

Comment:
both this method and the method below hide the TemporaryStreamStorage.  No one ever sees it except the handle object.    And it only sees it as an IDisposable.  None of the prior weird api where it was returned, with a large api, but you had to know what state it was in to use the api properly.